Summary of local setup for quickstart for devs (#372)
napari/docs: README.md

## contributing

Contributions are encouraged! Given that we're in an early stage, you may want to reach out on our [Github Issues](https://github.com/napari/docs/issues) before jumping in.
It's nice to be able preview how your contribution will look on the web, so here we provide a brief summary of what's required for a local setup that will allow you to do so.
Please read our [contributing guide](https://napari.org/dev/developers/contributing/documentation/index.html) for more comprehensive information about contributing documentation.

### quickstart: local setup
* __Check the prerequisites__

>     1. Create a [development installation](https://napari.org/stable/developers/contributing.html#dev-installation) of `napari` on your local machine.

Reviewer: ```suggestion
    1. Create a [development installation](https://napari.org/dev/developers/contributing/dev_install.html#dev-installation) of `napari` on your local machine.
```
Let's make sure we point to the latest dev docs.

    1. Fork this repository, and then clone your fork to your local machine. 

> NB: the name will be `napari-docs` rather than simply `docs`.

Reviewer: ```suggestion
NB: you may want to name your fork e.g. `napari-docs` rather than just `docs`.
```
Pretty sure *by default* it's called `docs` when you fork, but renaming it is a good idea IMO.

Author: Ah yeah I think I mixed up what happens when it's cloned vs. when it's forked, but I like this suggestion, I think it's nice to have the fork named that way anyhow.

    1. With your local clone's root folder as the working directory, and with the environment created in the first step activated, install the docs requirements with `python -m pip install -r requirements.txt`.
* __Build__ locally
    * If you're building on Windows, a few extra steps are required; you can follow [this guide](https://napari.org/stable/developers/documentation/index.html#building-the-documentation-on-windows).
    * From the root of your local clone of this repository, run one of these:

>         * `make docs`, if your changes include the example gallery

Reviewer 2: Should we mention `make docs-xvfb` here, or is that getting too detailed? 
It is certainly more convenient not to have napari windows popping up all over the place while it runs, but I guess if `xvfb` needs to be installed separately then perhaps this makes the instructions a little too long for the quickstart.

Reviewer: I *think* that only works on linux and if you have xvfb installed and set up?
I know I've fought with this in various docker/singularity containers.

Author: @GenevieveBuckley @psobolewskiPhD shall I add a note about this possibility, something like a bullet at the same indentation level as the Windows note, saying something like...

"If you're building on Linux and have `xvfb` set up, you can avoid windows opening during the build by using `make docs-xvfb`"

...or do you prefer as-is?

Reviewer: I think that's something that's best left for reading the actual docs. what's here is enough to get someone to a build and then they can explore the other `make` options

## contributing

> Contributions are encouraged! Given that we're in an early stage, you may want to reach out on our [Github Issues](https://github.com/napari/docs/issues) before jumping in.

Reviewer 3: I'm not sure I understand the comment about being in an early stage here - does this mean the contributor is just getting started, or that napari is in early stages? If it's the former maybe we want to say something like "Make sure you check our [issue tracker](https://github.com/napari/docs/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc) before jumping in."

Author: I guess I'm not sure, though I'd understood it as the latter. I left this sentence as it had been, but I added an additional sentence and reordered the sentences in this paragraph. I was not trying to change its meaning, sorry if the shuffling of sentences has made it more ambiguous 😅 happy to reword if preferred.

Reviewer: Yeah this sentence is a bit wierd! 
I think this may have been copy-pasta from the readme of the napari repo which used to be the home of the docs too, so then it made more sense?
Anyhow, I vote making it explicitly addressed to new contributors who happen upon the README vs. the actual docs on napari.org
How about: `If you're new or unsure about anything check out issues tracker or reach out on zulip`
https://napari.zulipchat.com

Author: OK, I reword [here](https://github.com/napari/docs/pull/372/commits/f704b39314d9fd8a4aefbdc723c5f519731a5015)

> It's nice to be able preview how your contribution will look on the web, so here we provide a brief summary of what's required for a local setup that will allow you to do so.
> Please read our [contributing guide](https://napari.org/dev/developers/contributing/documentation/index.html) for more comprehensive information about contributing documentation.
> 
> ### quickstart: local setup

Reviewer 3: We could also quickly mention the CircleCI here:

```suggestion
It's nice to be able preview how your contribution will look on the web, so here we provide a brief summary of what's required for a local setup that will allow you to do so. Alternatively, our CI set up provides a preview link that shows your changes. Please read our [contributing guide](https://napari.org/dev/developers/contributing/documentation/index.html) for more comprehensive information about contributing documentation.

### Quickstart: local setup
```

Author: I think if adding the CI setup note, the subheading should change to just "quickstart". Also, I think it should remain lowercase to match the rest of the README headings.

Reviewer: I like the idea of mentioning the easy CI preview somewhere, don't have an opinion where.

Author: I mention the CI preview [here](https://github.com/napari/docs/pull/372/commits/f704b39314d9fd8a4aefbdc723c5f519731a5015) and left the subheading as-is (both casing and text)

    1. Fork this repository, and then clone your fork to your local machine. NB: you may want to name your fork e.g. `napari-docs` rather than just `docs`.
    1. With your local clone's root folder as the working directory, and with the environment created in the first step activated, install the docs requirements with `python -m pip install -r requirements.txt`.
* __Build__ locally
    * If you're building on Windows, a few extra steps are required; you can follow [this guide](https://napari.org/stable/developers/documentation/index.html#building-the-documentation-on-windows).
    * From the root of your local clone of this repository, run one of these:
        * `make docs`, if your changes include the example gallery
        * `make docs-install && make html-noplot`, otherwise
* __Preview__ locally, either dragging-and-dropping `docs/_build/index.html` into a browser, or deploying a local server with `python3 -m http.server --directory docs/_build`.

These steps should set you up to build and preview your docs contributions on your local machine.

> For more detailed instructions and tips, please visit the relevant sections of our [contribution guide](https://napari.org/dev/developers/contributing/documentation/index.html).

Reviewer 3: I suggest only mentioning this once, instead of twice.
```suggestion
```

Author: @psobolewskiPhD @GenevieveBuckley WDYT? It sounds like there's a difference of opinion here.

Reviewer: 😬 
I think we can keep it, I kinda naturally started at quick-start....

Reviewer 3: Oh sorry I didn't see there was previous discussion about this. I'm happy to leave as is :D
